js/main.js

```diff
@@ -25,6 +23,90 @@ $form.addEventListener('submit', event => {
   data.entries.unshift(entry);
   data.nextEntryId++;
 
+  ul.prepend(renderEntry(entry));
+  viewSwap('entries');
+  toggleNoEntries();
+
   $photoPreview.setAttribute('src', 'images/placeholder-image-square.jpg');
   $form.reset();
 });
+
+// Render Entry Function
+function renderEntry(entry) {
+  const li = document.createElement('li');
+
+  const rowDiv = document.createElement('div');
+  rowDiv.className = 'row';
+  li.appendChild(rowDiv);
+
+  const imgContainerColumnHalfDiv = document.createElement('div');
+  imgContainerColumnHalfDiv.className = 'img-container column-half';
+  rowDiv.appendChild(imgContainerColumnHalfDiv);
+
+  const img = document.createElement('img');
+  img.setAttribute('src', entry.URL);
+  imgContainerColumnHalfDiv.appendChild(img);
+
+  const columnHalfDiv = document.createElement('div');
+  columnHalfDiv.className = 'column-half';
+  rowDiv.appendChild(columnHalfDiv);
+
+  const h3 = document.createElement('h3');
+  h3.textContent = entry.title;
+  columnHalfDiv.appendChild(h3);
+
+  const p = document.createElement('p');
+  p.textContent = entry.notes;
+  columnHalfDiv.appendChild(p);
+
+  return li;
+}
+
+// DOM Content Loaded Event Listener
+const ul = document.querySelector('ul');
+document.addEventListener('DOMContentLoaded', event => {
+  for (const entry of data.entries) {
+    ul.appendChild(renderEntry(entry));
+  }
+  viewSwap(data.view);
+  toggleNoEntries();
+});
+
+// Toggle No Entries Function
+const noEntriesMessage = document.querySelector('#no-entries-message');
+function toggleNoEntries() {
+  if (data.entries.length === 0) {
+    noEntriesMessage.classList.remove('hidden');
+  } else {
+    noEntriesMessage.classList.add('hidden');
+  }
+}
+
+// View Swap Function
+const entryFormview = document.querySelector('[data-view="entry-form"]');
+const entriesView = document.querySelector('[data-view="entries"]');
+function viewSwap(view) {
+
+  entryFormview.classList.add('hidden');
+  entriesView.classList.add('hidden');
```

Comment:
You can do this, but why hide both of them by default? Why not just hide them each individually in your conditional? If the view were to stay the same, we don't want to hide the page, we'd just want to keep it visible